odoo/modules/registry.py

```diff
@@ -467,15 +467,30 @@ def check_indexes(self, cr, model_names):
         if not expected:
             return
 
+        trgm = sql.has_pg_trgm(cr)
         cr.execute("SELECT indexname FROM pg_indexes WHERE indexname IN %s",
                    [tuple(row[0] for row in expected)])
         existing = {row[0] for row in cr.fetchall()}
 
-        for indexname, tablename, columnname, index in expected:
-            if index and indexname not in existing:
+        for indexname, model, columnname, field in expected:
+            tablename = model._table
+            index = field.index
+            assert index in ('btree', 'gin', 'not null', True, False)
+            if field.index and indexname not in existing:
+                where = ''
+                method = 'btree'
+                operator = ''
+                if index == 'not null':
```

Comment:
This is just lint really. It just looks weird, assign an intermediate variable index, don't use it in if statement, but then use it inside the if statement.